2014-02-20  Jason Merrill  <jason@redhat.com>

	PR c++/60274
	* g++.dg/template/ref7.C: New test.

--- gcc/testsuite/g++.dg/template/ref7.C	(revision 0)
+++ gcc/testsuite/g++.dg/template/ref7.C	(revision 207978)
@@ -0,0 +1,16 @@
+// PR c++/60274
+
+typedef const char *const&  ProtocolIdType;
+
+template <ProtocolIdType protocolId>
+struct C {
+  typedef int ProtocolVersion;
+  struct D {
+    ProtocolVersion GetProtocolVersion();
+  };
+};
+template <ProtocolIdType protocolId>
+typename C<protocolId>::ProtocolVersion C<protocolId>::D::GetProtocolVersion()
+{
+    return 1;
+}
